__tests__/reusable.test.js

```diff
@@ -0,0 +1,162 @@
+import {useState} from 'react'
+import {renderHook, act} from 'react-hooks-testing-library'
+import {getStore, createStore, replaceStore} from '../src/reusable';
+import {reusable, ReusableProvider} from '../src/react-reusable';
+
+let store;
+let unit;
+const value = {title: 'Yo'};
+const expectToBeAStore = (object) => {
+  expect(object.units).toBeDefined();
+}
+const expectToBeAUnit = (object) => {
+  expect(object.cachedValue).toBeDefined();
+}
+describe('public store API', () => {
+  beforeEach(() => {
+    store = getStore();
+  });
+  it('should allow to get a store', () => {
+    expectToBeAStore(store);
+  });
+  it('should allow to create a store', () => {
+    expectToBeAStore(createStore());
+  });
+  it('should allow to replace the store', () => {
+    const newStore = createStore();
+    replaceStore(newStore);
+    expect(getStore()).toBe(newStore);
+  });
+  it('should allow to create a unit', () => {
+    const fn = () => {};
+    store.createUnit(fn);
+    expectToBeAUnit(store.getUnit(fn));
+  });
+  it('should allow to subscribe to creating a unit', () => {
+    const fn = () => {};
+    let result;
+    store.onUnitsChanged(() => {
+      result = store.getUnit(fn);
+    });
+    store.createUnit(fn);
+    expectToBeAUnit(result);
+  });
+  it('should not allow to get a unit without creating it', () => {
+    const fn = () => {};
+
+    expect(() => store.getUnit(fn)).toThrow();
+  });
+  it('should not allow to create a unit twice', () => {
+    const fn = () => {};
+    store.createUnit(fn);
+    expect(() => store.createUnit(fn));
+  });
+  it('should allow to unsubscribe from subscribing to creating a unit', () => {
+    expect.assertions(0);
```

Comment:
why not pass a jest spy to store.onUnitsChanged and then expect(spy).not.toHaveBeenCalled ?